src/problem2.rb

```diff
@@ -0,0 +1,46 @@
+class Problem2
+  def calculateMaximum(values) #theta(n/2) -> O(n)
```

Comment:
A good practice for comments (unless used for documentation) is to denote `why`, not `what`.